_posts/2018-01-22-Java9-modules-Spring-Boot-2-Docker.md

```diff
@@ -0,0 +1,301 @@
+---
+layout: post
+author: arto
+title: Put your Java on a diet with Java 9 Modules - Spring Boot 2 - Docker - Oh my!
+excerpt: New year, new technologies. Java 9 modules, Spring Boot 2 and Docker work extremely well together, and I'll show you how.
+tags:
+- Java
+- Java 9
+- Spring
+- Spring Boot
+- Microservices
+- Docker
+---
+
+It's year 2018, and it's time to update your knowledge to date. Many people are running Java 8 or older, yet Java 8 will go EOL by September (and anything older is already extinct). This means no security updates for you, well at least if you don't pay for them. Furthermore, Java 8 was released in 2014, based on work done before that, so if you are working on Java 8 or older platforms, you are using technology that is very old by todays pace, and limiting your options severely.
+
+You will need to move on to Java 9 at some point. The earlier you can start flirting with it the easier it will be for you. I like to work at cutting edge myself, to embrace challenges at early date, and to evaluate new advances a bit early. Then new things become tools in my toolkit, so I can apply them when they do make the most sense. In some places, yes, it makes sense to run older versions, that are stable and have had many bugfixes and updates already. You might even feel it's impossible to move on to new versions. Perhaps so, but todays technologies like Docker make it easier, and if you think you cannot update your platforms now and then, cannot find a way to do so, you might be the worst obstacle for updating yourself. There is often a way, if there's a will. With todays technologies, almost anything is possible.
+
+I played a lot with JDK 9 prereleases, betas, early access, to get hold of new things like the module system, jshell, new stream API tricks, etc. But now it's out of beta, general availability, with some updates in already. Ready to go.
+
+But it's not going to be easy. Java 9 is going to be a huge step due to its module systems and new deprecations. It will simply not run many of your existing software stacks and libraries. For example, Spring Boot 1.x is not going to work at all, and it's not going to be working at any future point either. If you wish to run Spring Boot, like many do, you need to first update Spring Boot to version 2.0. Many other libraries will break. Some have been updated, some will be updated at some point, some never will. What will help you a lot with experimentation, is Docker.
+
+## Docker
+
```

Comment:
"For production it has much more different qualities, but they are not the topic today." Jättäisin tuon kokonaan pois. Pitkä kirjoitus joka tapauksessa.